src/pykeen/nn/init.py

```diff
@@ -369,6 +386,8 @@ def __call__(self, x: torch.Tensor) -> torch.Tensor:
         x[:] = color_representation[self.colors]
         return x
 
+    # TODO as_embedding function
```

Comment:
it has the one derived from `PretrainedInitializer`

Reply:
This doesn’t inherit from that class

Reply:
Can we make it?

Reply:
`color_initializer` can have a random component (and iirc, the default initializer is randomized).

Thus, it is not a `PretrainedInitializer`; I just misread the collapsed diff 😅

Reply:
but
```python
x = Embedding(max_id=..., shape=shape, initializer=initializer, trainable=False)
```
is not much more effort to write than
```python
x = initializer.as_embedding()
```
so this may be fine 🙂

Reply:
Is there a way to infer the max ID from this initializer? For sake of consistency I think it would be nice to make this work

Reply:
yes,
```python
self.colors.shape[0]
```

Reply:
Reconsidering, I think colors could also be a `TokenizationRepresentation` with `num_tokens=1` 🤔

Reply:
In db0373e9 i made it inherit from the `PretrainedInitializer`

Reply:
Hm, `colors` is `torch.long` iirc, and of shape `(max_id,)`. And the actual "embedding" of shape `(max_id, dim)` then gets constructed from that in the `__call__` 🤔

Reply:
can we have it constructed in the `__init__` instead to keep the same interface? I don't think I quite understand this one

Reply:
We could, but then we lose the stochasticity of the color initializer.

This initializer consists of two steps:
1. it calculates categorical WL colors for each entity. this is done once in the `__init__`.
2. it encodes the categorical colors to vectors. This is done whenever we `__call__`. The default here is to use randomly generated color embeddings for each categorical, i.e., for each color a fixed-size vector sampled from a standard normal distribution. This is inspired by https://arxiv.org/abs/2001.09621, Section 3.4, "Replacing node indicators functions".